nox/_decorators.py

```diff
@@ -1,7 +1,7 @@
 import copy
 import functools
 import types
-from typing import Any, Callable, Dict, Iterable, List, Optional, cast
+from typing import AbstractSet, Any, Callable, Dict, Iterable, List, Optional, cast
```

Comment:
Very minor nitpick: note that `AbstractSet` is deprecated [in 3.9](https://docs.python.org/3/library/typing.html#typing.AbstractSet)

Reply:
Thank you for taking the time to review this.

`typing.AbstractSet` is superceded by `collections.abc.Set` in 3.9, but this won't work in earlier Python versions where the abstract collections are not yet generics, i.e. you can't write `Set[str]`. The same deprecation affects Callable and Iterable; Dict and List are also deprecated in favor of `dict` and `list`.

I'm not sure how to address this, do you have an idea? It would be interesting to see how other projects deal with the deprecation. I'm hesitant to introduce a large amount of compatibility boilerplate.

Reply:
For what's worth, in `pytest` the contract when `ids` are explicitly provided is to receive an `Iterable[str]` (and to transform it into a list or set internally so as to consume possible generators). Maybe this could be sufficient ?

Reply:
The public API does use `Iterable[str]`:

```python
# _parametrize.py
def parametrize_decorator(
    arg_names: Union[str, List[str], Tuple[str]],
    arg_values_list: Union[Iterable[ArgValue], ArgValue],
    ids: Optional[Iterable[Optional[str]]] = None,
    sessionparams: Optional[Iterable[str]] = None,
) -> Callable[[Any], Any]:
    ...
    sessionparams = frozenset(sessionparams if sessionparams is not None else [])
    ...
```

We only use `AbstractSet` internally.

Reply:
oh sorry I read too fast ! perfect then.